admin-api-server/src/analytics/service/analytics.service.ts

```diff
@@ -46,32 +49,72 @@ export class AnalyticsService {
   ): Promise<MetricEntity[]> {
     const qryRes = await this.#timeseries(params);
 
-    return qryRes.rows.map(
-      (row: any) =>
-        <MetricEntity>{
-          timestamp: Math.floor(row['timestamp'].getTime() / 1000),
-          value: Number(row['nft_count']),
-        },
-    );
+    const nftCountMap: Record<string, number> = {};
+    qryRes.rows.forEach((row: any) => {
+      const key = Math.floor(row['timestamp'].getTime() / 1000).toString();
+      nftCountMap[key] = Number(row['nft_count']);
+    });
+
+    return this.getTimestampData(nftCountMap, params.resolution);
   }
 
   async getTimeseriesSalesPriceVolume(
     params: MetricParams,
   ): Promise<MetricEntity[]> {
     const qryRes = await this.#timeseries(params);
 
-    return qryRes.rows.map(
-      (row: any) =>
-        <MetricEntity>{
-          timestamp: Math.floor(row['timestamp'].getTime() / 1000),
-          value: Number(
-            this.currencyService.convertToCurrency(
-              Number(row['price_volume']),
-              BASE_CURRENCY,
-            ),
-          ),
-        },
-    );
+    const priceVolumeMap: Record<string, number> = {};
+    qryRes.rows.forEach((row: any) => {
+      const key = Math.floor(row['timestamp'].getTime() / 1000).toString();
+
+      priceVolumeMap[key] = Number(
+        this.currencyService.convertToCurrency(
+          Number(row['price_volume']),
+          BASE_CURRENCY,
+        ),
+      );
+    });
+
+    return this.getTimestampData(priceVolumeMap, params.resolution);
+  }
+
+  getTimestampData(
```

Comment:
This name can be improved. It's not indicating well enough what it does. Perhaps something like: `zeroFillTimeSeries`?

Reply:
Agree, I usually like verb names but I find `getZeroFilledTimeseries` too long so your idea is much better